hedera-mirror-rest/__tests__/service/nftService.test.js

```diff
@@ -20,39 +20,312 @@
+          {key: TOKEN_ID, operator: lt, value: 10},
+        ],
+      },
+      expected: {
+        sqlQuery: `(${selectColumnsStatement}
+              from nft where account_id = $1
+              and token_id >= $3
+              and token_id < $4
+              order by token_id desc,serial_number desc limit $2
+            ) union all (
+              ${selectColumnsStatement}
+              from nft where account_id = $1
+              and token_id = $5
+              and serial_number < $6
+              order by token_id desc,serial_number desc limit $2
+            )
+            order by token_id desc,serial_number desc
+            limit $2`,
+        params: [1, 20, 3, 10, 50, 30],
+      },
+    },
+    {
+      name: 'token closed range and serial closed range',
+      query: {
+        ...defaultQuery,
+        lower: [
+          {key: TOKEN_ID, operator: eq, value: 2},
+          {key: SERIAL_NUMBER, operator: gte, value: 100},
+        ],
+        inner: [
+          {key: TOKEN_ID, operator: gt, value: 2},
+          {key: TOKEN_ID, operator: lt, value: 8},
+        ],
+        upper: [
+          {key: TOKEN_ID, operator: eq, value: 8},
+          {key: SERIAL_NUMBER, operator: lte, value: 200},
+        ],
+      },
+      expected: {
+        sqlQuery: `(${selectColumnsStatement}
+              from nft where account_id = $1
+              and token_id = $3
+              and serial_number >= $4
+              order by token_id desc,serial_number desc limit $2
+            ) union all (
+              ${selectColumnsStatement}
+              from nft where account_id = $1
+              and token_id > $5
+              and token_id < $6
+              order by token_id desc,serial_number desc limit $2
+            ) union all (
+              ${selectColumnsStatement}
+              from nft where account_id = $1
+              and token_id = $7
+              and serial_number <= $8
+              order by token_id desc,serial_number desc limit $2
+            )
+            order by token_id desc,serial_number desc
+            limit $2`,
+        params: [1, 20, 2, 100, 2, 8, 8, 200],
+      },
+    },
+    {
+      name: 'spender eq in gte lte',
+      query: {
+        ...defaultQuery,
+        spenderIdInFilters: [
+          {key: SPENDER_ID, operator: eq, value: 15},
+          {key: SPENDER_ID, operator: eq, value: 17},
+          {key: SPENDER_ID, operator: eq, value: 22},
+        ],
+        spenderIdFilters: [
+          {key: SPENDER_ID, operator: lte, value: 10},
+          {key: SPENDER_ID, operator: gte, value: 30},
+        ],
+      },
+      expected: {
+        sqlQuery: `${selectColumnsStatement}
+            from nft where account_id = $1
+            and spender <= $3
+            and spender >= $4
+            and spender in (15,17,22)
+            order by token_id desc,serial_number desc
+            limit $2`,
+        params: [1, 20, 10, 30],
```

Comment:
as mentioned in `nftService.js`, we should generate such a SQL query
```suggestion
            and spender in ($5,$6,$7)
            order by token_id desc,serial_number desc
            limit $2`,
        params: [1, 20, 10, 30, 15, 17, 22],
```

Reply:
I have updated the tests to reflect the inclusion of the values in the params.